packages/flutter_tools/gradle/src/main/groovy/shared.groovy

```diff
@@ -0,0 +1,114 @@
+            // packages/flutter_tools/lib/src/flutter_plugins.dart.
+            assert androidPlugin.name instanceof String
+            assert androidPlugin.path instanceof String
+            assert androidPlugin.dependencies instanceof List<String>
+            // Skip plugins that have no native build (such as a Dart-only implementation
+            // of a federated plugin).
+            def needsBuild = androidPlugin.containsKey(nativeBuildKey) ? androidPlugin[nativeBuildKey] : true
+            if (needsBuild) {
+                nativePlugins.add(androidPlugin)
+            }
+        }
+        return nativePlugins
+    }
+
+
+    private Map<String, Object> parsedFlutterPluginsDependencies
+
+    /**
+     * Parses <project-src>/.flutter-plugins-dependencies
+     */
+    Map<String, Object> getDependenciesMetadata(File flutterSourceDirectory) {
+        // Consider a `.flutter-plugins-dependencies` file with the following content:
+        // {
+        //     "plugins": {
+        //       "android": [
+        //         {
+        //           "name": "plugin-a",
+        //           "path": "/path/to/plugin-a",
+        //           "dependencies": ["plugin-b", "plugin-c"],
+        //           "native_build": true
+        //         },
+        //         {
+        //           "name": "plugin-b",
+        //           "path": "/path/to/plugin-b",
+        //           "dependencies": ["plugin-c"],
+        //           "native_build": true
+        //         },
+        //         {
+        //           "name": "plugin-c",
+        //           "path": "/path/to/plugin-c",
+        //           "dependencies": [],
+        //           "native_build": true
+        //         },
+        //       ],
+        //     },
+        //     "dependencyGraph": [
+        //       {
+        //         "name": "plugin-a",
+        //         "dependencies": ["plugin-b","plugin-c"]
+        //       },
+        //       {
+        //         "name": "plugin-b",
+        //         "dependencies": ["plugin-c"]
+        //       },
+        //       {
+        //         "name": "plugin-c",
+        //         "dependencies": []
+        //       }
+        //     ]
+        // }
+        // This means, `plugin-a` depends on `plugin-b` and `plugin-c`.
+        // `plugin-b` depends on `plugin-c`.
+        // `plugin-c` doesn't depend on anything.
+        if (parsedFlutterPluginsDependencies) {
+            return parsedFlutterPluginsDependencies
+        }
+        File pluginsDependencyFile = new File(flutterSourceDirectory, flutterPluginsDependenciesFile)
+        if (pluginsDependencyFile.exists()) {
+            def object = new JsonSlurper().parseText(pluginsDependencyFile.text)
+            assert object instanceof Map<String, Object>
+            parsedFlutterPluginsDependencies = object
+            return object
+        }
+        return null
+    }
+}
+
+// TODO(135392): Remove and use declarative form when migrated
```

Comment:
Is this your todo or a prexisting one? if it is yours can you use the github shorthand for an issue flutter/flutter/issues/135392, If it is not new then dont worry about making the change.

Reply:
It is a new TODO from my side. Can you state the exact syntax? I did not find a similar TODO by searching the repo. I used the syntax proposed [here](https://github.com/flutter/flutter/pull/137115#discussion_r1388583026).
Or do you mean I should link this in the according issue?

Reply:
I mean that 135392 is harder to reason about than either "flutter/flutter/issues/135392" or github.com/ url to the issue. That said the analyzer is more important to make happy than me.

Reply:
Attempt in ae467f821f2b7bfcab82fbf338e79d5779fba4a9

Reply:
Unfortunately the linter complains :/